### PR TITLE
[DO NOT LAND] Upgrade inspector_protocol

### PR DESCRIPTION
**Pull request by kfarnung**
* Upgrade inspector_protocol to
  c22d4bd88fb7a39bc41c3b1adcdd733cc9b5e8ea
* Switch over to the standard PDL file plus TTD
* Stub out new functions to get everything building

This is an initial attempt to see what differences there are with the latest protocol, **do not land**.

### CODE REVIEW

**Comment by akrantz** [comment]
Many of these changes seem to be differences in the generation rather than changes to the underlying protocol. Is there a way to see the diff of the protocol itself?

**Comment by akrantz** [comment]
It looks like there is a python script to write out json for the inspector protocol. It would be nice to know what the diff looks like between versions.

**Comment by kfarnung** [comment]
@akrantz We already run that script as part of the regeneration, the diff looks pretty bad right now since the old file wasn't generated.  The easiest way to see the command differences is to look at the implementation changes in the ProtocolHandler.  It's best to ignore the generated files for what you're looking for.

Going forward it seems like the PDL is a much better format for reviewing changes.

File: lib/Debugger.Protocol/js_protocol.pdl [hunk 1]
```diff
@@ -0,0 +1,1468 @@
...
+
+  # Returns supported domains.
+  command getDomains
+    returns
+      # List of supported domains.
+      array of Domain domains
+
+# TimeTravel domain exposes JavaScript time travel capabilities. It allows
+# stepping backwards through execution.
+domain TimeTravel
```
Comment (kfarnung) [inline lib/Debugger.Protocol/js_protocol.pdl]: Added our TimeTravel domain here, but it's not currently enabled.

File: lib/Debugger.ProtocolHandler/DebuggerCallFrame.cpp [hunk 2]
```diff
@@ -114,6 +114,7 @@ namespace JsDebug
             .setReturnValue(GetReturnValue())
             .setScopeChain(GetScopeChain())
             .setThis(GetThis())
+            .setUrl("")
```
Comment (kfarnung) [inline lib/Debugger.ProtocolHandler/DebuggerCallFrame.cpp]: This is now required, but the data isn't readily available at this point in the code.

File: lib/Debugger.ProtocolHandler/DebuggerImpl.h [hunk 3]
```diff
@@ -25,73 +25,96 @@ namespace JsDebug
         DebuggerImpl& operator=(const DebuggerImpl&) = delete;
 
         // protocol::Debugger::Backend implementation
-        protocol::Response enable() override;
```
Comment (kfarnung) [inline lib/Debugger.ProtocolHandler/DebuggerImpl.h]: I reordered the implemented methods alphabetically to make adding them easier.  The order is unchanged in the implementation file.

File: lib/Debugger.ProtocolHandler/RuntimeImpl.h [hunk 4]
```diff
@@ -21,32 +21,48 @@ namespace JsDebug
         RuntimeImpl& operator=(const RuntimeImpl&) = delete;
 
         // protocol::Runtime::Backend implementation
-        void evaluate(
```
Comment (kfarnung) [inline lib/Debugger.ProtocolHandler/RuntimeImpl.h]: I reordered the implemented methods alphabetically to make adding them easier. The order is unchanged in the implementation file.

**Comment by akrantz** [comment]
@kfarnung Closing this pull request since it seems to have been experimental.

**Pull request closed without merging**